### DISCUSSION TITLE
Why cant I import  kanren.constraints and  kanren.graph

### DISCUSSION THREAD

**Discussion by wirtschaftswunder** [pythological/kanren#83] [General]
Hello everyone,

Did install minikanren on Ubuntu 18(bionic) system via pyenv and first examples work (Python 3.9.10) but when I try to import from 
kanren.constraints and  kanren.graph I m only getting No module named 'kanren.graph error .
Install was done via pip and believe latest version is used (1.0.3) 
Was there something removed as dir(kanren) doesn't shows any constraints or graph 

Rgds 

Gottfried

**Comment by brandonwillard** [comment]
If you can provide an [MWE](https://en.wikipedia.org/wiki/Minimal_reproducible_example) that reproduces the environment in which you're working, I can take a look.

**Reply by wirtschaftswunder** [reply]
Brandon,
I did it again with pip install miniKanren==1.0.2 and it is working but it does not with latest version and following command pip install miniKanren

It starts with import error for etuples but when I repeat import command it is accepted and also goals and Relation import correct 
Only graph and constraints show no module error 

>>> from etuples.core import etuple, ExpressionTuple
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/goenzoy/.pyenv/versions/kanren/lib/python3.9/site-packages/etuples/__init__.py", line 3, in <module>
    from .dispatch import apply, arguments, etuplize, operator, rands, rator, term
  File "/home/goenzoy/.pyenv/versions/kanren/lib/python3.9/site-packages/etuples/dispatch.py", line 3, in <module>
    from cons.core import ConsError, ConsNull, ConsPair, car, cdr, cons
  File "/home/goenzoy/.pyenv/versions/kanren/lib/python3.9/site-packages/cons/__init__.py", line 1, in <module>
    from . import unify  # noqa: F401
  File "/home/goenzoy/.pyenv/versions/kanren/lib/python3.9/site-packages/cons/unify.py", line 4, in <module>
    from unification.core import _reify, _unify, construction_sentinel
ImportError: cannot import name 'construction_sentinel' from 'unification.core' (/home/goenzoy/.pyenv/versions/kanren/lib/python3.9/site-packages/unification/core.py)
>>> from etuples.core import etuple, ExpressionTuple
>>> from kanren.goals import conso
>>> from kanren import Relation, facts
>>> from kanren.graph import walko
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'kanren.graph'
>>> 

If I do the same with 1.0.2 it is fine like below 

Successfully installed miniKanren-1.0.2
WARNING: You are using pip version 21.2.4; however, version 23.3.1 is available.
You should consider upgrading via the '/home/goenzoy/.pyenv/versions/3.9.10/envs/kanren2/bin/python3.9 -m pip install --upgrade pip' command.
(kanren2) XXXXXXXXXXX:~$ python         (just blanked host-name)  
Python 3.9.10 (main, Jul 29 2023, 23:55:02) 
[GCC 7.5.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from etuples.core import etuple, ExpressionTuple
>>> from kanren import Relation, facts
>>> from kanren.goals import conso
>>> from kanren.graph import walko
>>> from functools import reduce
>>> import csv
>>> from kanren.constraints import neq, isinstanceo
>>>
